Pokemon/Pokemon/PKPokedexTableViewController.m

```diff
@@ -0,0 +1,481 @@
+    } else {
+        return nil;
+    }
+}
+
+- (void) prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender{
+    
+    NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
+    
+    if (self.showAllPokemon){
+        NSString *pokemonName = self.allPokemonAtoZ [indexPath.row];
+        PKPokemonClickDetailsViewController *detailView = segue.destinationViewController;
+        detailView.pokemonName = pokemonName;
+    } else {
+        NSArray *keys = [self.pokemon allKeys];
+        NSString *key = keys[indexPath.section];
+        NSArray *pokemonArray = [self.pokemon objectForKey:key];
+        NSString *pokemonName = pokemonArray[indexPath.row];
+        PKPokemonClickDetailsViewController *detailView = segue.destinationViewController;
+        detailView.pokemonName = pokemonName;
+    }
+    
+    
+}
+
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    if (self.showAllPokemon){
+        return 1;
+    } else {
+    NSArray *keys = [self.pokemon allKeys];
+    return keys.count;
+    }
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    
+    if (self.showAllPokemon){
+        return [self.allPokemonAtoZ count];
+    } else {
+    NSArray *keys = [self.pokemon allKeys];
+    NSString *key = keys[section];
+    NSArray *values = self.pokemon[key];
+    return values.count;
+    }
+}
+
+//- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+//    NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
+//    
+//    NSArray * keys = [self.pokemonList allKeys];
+//    NSString *key = keys[indexPath.section];
+//    NSArray * pokemonArray = [self.pokemonList objectForKey:key];
+//    NSString * pokemonName = pokemonArray[indexPath.row];
+//    PokemonFactsViewController *vc = segue.destinationViewController;
+//    vc.pokemonName = pokemonName;
+//}
+//
+//- (void) prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+//
+//    NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
+//    
+//    NSArray *keys = [self.pokemon allKeys];
+//    NSString *key = keys[indexPath.section];
+//    NSArray *pokemonArray = [self.pokemon objectForKey:key];
+//    NSString *pokemonName = pokemonArray[indexPath.row];
+//    
+//    PKPokemonClickDetailsViewController *vc = segue.destinationViewController;
+//    vc.pokemonName = pokemonName;
+//    
+//
+//}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"reuseIdentifier" forIndexPath:indexPath];
+    
+    if (self.showAllPokemon){
+        NSString *pokemonName = self.allPokemonAtoZ[indexPath.row];
+        cell.textLabel.text = pokemonName;
+        cell.imageView.image = [UIImage imageNamed:pokemonName];
+        return cell;
+    } else {
+    
+    NSArray *keys = [self.pokemon allKeys];
```

Comment:
Everything within the else { } conditions should be indented the same as the code in the if () { } block.  This is true for all your if/else conditions.

For example:

if (self.showAllPokemon) {
  somecode;
} else {
  somemorecode;
}